### DISCUSSION TITLE
How to setup asset discovery for Linux Arm64

### DISCUSSION THREAD

**Discussion by GProst** [percy/cli#672] [Support] [answered]
## The problem

Trying to run Percy with Capybara on CercleCI (Ubuntu 20.04 arm64).
IIUC percy installs some local Chromium browser, after browser download is completed percy doesn't start due to the following error:
```
[percy] Skipping visual tests - Failed to launch browser. 
[...]/node_modules/@percy/core/.local-chromium/885264/chrome-linux/chrome: 17: Syntax error: Unterminated quoted string
```
Tests run but snapshots aren't taken, each time logs the following: `[percy] Percy is not running, disabling snapshots`

We use remote browser there to run Capybara tests, can't Percy just use remote browser instead of its local one?

## Environment

- Node version: 12.22.7
- `@percy/cli` version: 1.0.0-beta.71
- Version of Percy SDK you’re using: percy-capybara 5.0.0
- OS version: Ubuntu 20.04 arm64

**Comment by Robdel12** [comment, marked as answer]
Hey @GProst! At first I thought this was a bug because `@percy/core` should download the appropriate Chromium browser for the machine that the SDK is running on. We added M* Mac support in the past (#486), so I figured the same needed to be done here for arm64 linux. 

Unfortunately, I think I figured out why Puppeteer also has this issue. It looks like there [aren't any prebuilt revisions of Chromium for arm linux](https://storage.googleapis.com/chromium-browser-snapshots/index.html) newer than 2014. Most Puppeteer issues suggest installing chrome and then setting the executable path to that installed Chrome (`sudo apt install chromium-browser chromium-codecs-ffmpeg`). That's what will have to be done, since there aren't any prebuilt versions offered.

Once you have the browser installed in CI, you can then tell the Percy SDK to use that browser by setting the executable path https://docs.percy.io/docs/skipping-asset-discovery-browser-download

Why we need a browser running in CI is described here: https://docs.percy.io/docs/debugging-sdks#asset-discovery
